volatility3/framework/configuration/__init__.py

```diff
@@ -3,3 +3,5 @@
 #
 
 from volatility3.framework.configuration import requirements
+
+__all__ = ["requirements"]
```

Comment:
I'm wary about wanting to add a new file and then forgetting it need adding here too.  What was this specific rule that flagged this?

Reply:
Sure, but there's a few places where we expect the resolver to pick them up based on the file structure.  If we include `__all__` then that resolver never gets involved and bits can fall apart (which is why I don't want to set it as a precedent everywhere).

Reply:
what's the resolver? :sweat_smile:

Reply:
Just if you ask python to `import blah.foo` it'll look up foo as a file under `blah` and try to load it, unless blah has an `__all__` that doesn't include it?

Reply:
Hmmm, I'm happy with always using `as` to reduce module name length (although we have some modules named the same, so `context` and `interfaces.context`, but I can live with `interfaces_context` for clarity.  We also work hard to prevent people doing `import *` of anything, but adding a new file to the `__all__` list feels like an extra step we might forget.  I'd go so far as to ok that rule being added to the ignore list across the whole codebase, as long as it wouldn't allow anything else through?

Reply:
the problem i see is that catching unused imports is well an important thing, ingoring it in all `__init__.py`s is what i'd suggest if the `__all__` stuff isn't possible/wanted

Reply:
I don't think that's a significant problem?  The style guide ensures that most imports are specific (down to naming the classes to import) and where we do import a full module without specifying the individual submodules we want, it's probably because we want everything (or we don't know yet what we're going to want).  In either care, we'll likely end up asking for everything, and if we then *don't* get it because someone forgot to include the new file in a variable, it causes friction.  It feels almost like duplication, in that you're explicitly stating the modules, as well as putting them in the directory?  Shouldn't putting them in the directory be enough to have them included?

Reply:
removed here: https://github.com/volatilityfoundation/volatility3/pull/1364/commits/f2771bce87e80c7a988247f855e5afa117753654